main.nf

```diff
@@ -197,32 +197,33 @@ params.snpeff_db               = params.genome ? params.genomes[params.genome].s
+chr_dir           = params.chr_dir           ? file(params.chr_dir)           : file("${params.outdir}/no_file")
+chr_length        = params.chr_length        ? file(params.chr_length)        : file("${params.outdir}/no_file")
+dbsnp             = params.dbsnp             ? file(params.dbsnp)             : file("${params.outdir}/no_file")
+fasta             = params.fasta             ? file(params.fasta)             : file("${params.outdir}/no_file")
+germline_resource = params.germline_resource ? file(params.germline_resource) : file("${params.outdir}/no_file")
+known_indels      = params.known_indels      ? file(params.known_indels)      : file("${params.outdir}/no_file")
+loci              = params.ac_loci           ? file(params.ac_loci)           : file("${params.outdir}/no_file")
+loci_gc           = params.ac_loci_gc        ? file(params.ac_loci_gc)        : file("${params.outdir}/no_file")
+mappability       = params.mappability       ? file(params.mappability)       : file("${params.outdir}/no_file")
 
 // Initialize value channels based on params, defined in the params.genomes[params.genome] scope
 snpeff_db         = params.snpeff_db         ?: Channel.empty()
 snpeff_species    = params.species           ?: Channel.empty()
 vep_cache_version = params.vep_cache_version ?: Channel.empty()
 
 // Initialize files channels based on params, not defined within the params.genomes[params.genome] scope
-cadd_indels       = params.cadd_indels       ?: Channel.empty()
-cadd_indels_tbi   = params.cadd_indels_tbi   ?: Channel.empty()
-cadd_wg_snvs      = params.cadd_wg_snvs      ?: Channel.empty()
-cadd_wg_snvs_tbi  = params.cadd_wg_snvs_tbi  ?: Channel.empty()
-pon               = params.pon               ?: Channel.empty()
-snpeff_cache      = params.snpeff_cache      ?: Channel.empty()
-//ch_target_bed = params.target_bed ? Channel.value(file(params.target_bed)) : "null"
-target_bed        =  params.target_bed        ?: Channel.empty()
-vep_cache         = params.vep_cache         ?: Channel.empty()
+cadd_indels       = params.cadd_indels       ? file(params.cadd_indels)      : file("${params.outdir}/no_file")
+cadd_indels_tbi   = params.cadd_indels_tbi   ? file(params.cadd_indels_tbi)  : file("${params.outdir}/no_file")
+cadd_wg_snvs      = params.cadd_wg_snvs      ? file(params.cadd_wg_snvs)     : file("${params.outdir}/no_file")
+cadd_wg_snvs_tbi  = params.cadd_wg_snvs_tbi  ? file(params.cadd_wg_snvs_tbi) : file("${params.outdir}/no_file")
+pon               = params.pon               ? file(params.pon)              : file("${params.outdir}/no_file")
+snpeff_cache      = params.snpeff_cache      ? file(params.snpeff_cache)     : file("${params.outdir}/no_file")
+target_bed        = params.target_bed        ? file(params.target_bed)       : file("${params.outdir}/no_file")
+vep_cache         = params.vep_cache         ? file(params.vep_cache)        : file("${params.outdir}/no_file")
 
```

Comment:
Does the no_file folder then remain empty?

Reply:
I'm actually creating the `no_file` file a couple of lines before with:
```
file("${params.outdir}/no_file").text = "no_file\n"
```

Reply:
Ah ok, so actually just an empty file is used then?